File: app/src/main/java/com/github/batulovandrey/unofficialurbandictionary/adapter/DefinitionAdapter.java
package com.github.batulovandrey.unofficialurbandictionary.adapter;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.github.batulovandrey.unofficialurbandictionary.R;
import com.github.batulovandrey.unofficialurbandictionary.bean.DefinitionResponse;

import java.util.List;

/**
 * @author Andrey Batulov on 26/10/2017
 */

public class DefinitionAdapter extends RecyclerView.Adapter<DefinitionViewHolder> {

    private List<DefinitionResponse> mDefinitions;
    private DefinitionClickListener mClickListener;

    public DefinitionAdapter(List<DefinitionResponse> definitions,
                             DefinitionClickListener clickListener) {
        mDefinitions = definitions;
        mClickListener = clickListener;
    }

    @Override
    public DefinitionViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.definition_item, null);
        return new DefinitionViewHolder(view, mClickListener);
    }

    @Override
    public void onBindViewHolder(DefinitionViewHolder holder, int position) {
        DefinitionResponse definition = mDefinitions.get(position);
        holder.mDefinitionTextView.setText(definition.getDefinition());
        holder.mWordTextView.setText(definition.getWord());
        holder.mExampleTextView.setText(definition.getExample());
        holder.mAuthorTextView.setText(definition.getAuthor());
    }

    @Override
    public int getItemCount() {
        return mDefinitions.size();
    }
}

File: app/src/main/java/com/github/batulovandrey/unofficialurbandictionary/adapter/DefinitionAdapter.kt
package com.github.batulovandrey.unofficialurbandictionary.adapter

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.ViewGroup
import com.github.batulovandrey.unofficialurbandictionary.R
import com.github.batulovandrey.unofficialurbandictionary.data.db.model.Definition

/**
 * @author Andrey Batulov on 22/12/2017
 */

class DefinitionAdapter(private val definitions: List<Definition>,
                        private val clickListener: DefinitionClickListener) : RecyclerView.Adapter<DefinitionViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): DefinitionViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.definition_item, null)
        return DefinitionViewHolder(view, clickListener)
    }

    override fun onBindViewHolder(holder: DefinitionViewHolder, position: Int) {
        val definition = definitions[position]
        holder.setDefinitionText(definition.definition)
        holder.setAuthorText(definition.author)
        holder.setExampleText(definition.example)
        holder.setWordText(definition.word)
    }

    override fun getItemCount(): Int {
        return definitions.size
    }

    fun getDefinitionByPosition(position: Int) = definitions[position]
}